cmd/system-probe/subcommands/usm/netstat_linux.go

```diff
@@ -0,0 +1,336 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package usm
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/command"
+	sysconfigcomponent "github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/procnet"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+const (
+	// /proc/net/* file field indices
+	procNetFieldLocalAddr  = 1
+	procNetFieldRemoteAddr = 2
+	procNetFieldState      = 3
+	procNetFieldInode      = 9
+	procNetMinFields       = 10
+
+	// IPv4 and IPv6 hex address lengths
+	ipv4HexLength = 8
+	ipv6HexLength = 32
+)
+
+// TCP connection states (from include/net/tcp_states.h)
+const (
+	tcpStateEstablished = iota + 1
+	tcpStateSynSent
+	tcpStateSynRecv
+	tcpStateFinWait1
+	tcpStateFinWait2
+	tcpStateTimeWait
+	tcpStateClose
+	tcpStateCloseWait
+	tcpStateLastAck
+	tcpStateListen
+	tcpStateClosing
+)
+
+// makeNetstatCommand returns the "usm netstat" cobra command
+func makeNetstatCommand(globalParams *command.GlobalParams) *cobra.Command {
+	var showTCP bool
+	var showUDP bool
+	var showListening bool
+
+	cmd := makeOneShotCommand(
+		globalParams,
+		"netstat",
+		"Show network connections similar to netstat -antpu",
+		func(_ sysconfigcomponent.Component, _ *command.GlobalParams) error {
+			return runNetstat(showTCP, showUDP, showListening)
+		},
+	)
+
+	cmd.Flags().BoolVarP(&showTCP, "tcp", "t", true, "Show TCP connections")
+	cmd.Flags().BoolVarP(&showUDP, "udp", "u", true, "Show UDP connections")
+	cmd.Flags().BoolVarP(&showListening, "listening", "l", false, "Show only listening sockets")
```

Comment:
I think we can remove this option, and the additional code for it
running `system-probe usm netstat | grep -i listening` yields the same output

Reply:
since `--listening` is just doing a filtering, you can use grep for it
`grep -i -e listening -e udp`

The point is that you don't need to actually implement that filtering logic since you can use grep
But that's up to you